src/crimson/os/seastore/cache.cc

```diff
@@ -1175,7 +1176,13 @@ void Cache::complete_commit(
     bool is_inline = false;
     if (i->is_inline()) {
       is_inline = true;
-      i->set_paddr(final_block_start.add_relative(i->get_paddr()));
+      if (final_block_start.get_addr_type() == addr_types_t::SEGMENT) {
+	i->set_paddr(final_block_start.add_relative(i->get_paddr()));
+      } else if (final_block_start.get_addr_type() ==
+	addr_types_t::RANDOM_BLOCK) {
+	i->set_paddr(final_block_start.add_offset(
+	  i->get_paddr().as_seg_paddr().get_segment_off()));
+      }
```

Comment:
Same comment, the goal paddr_t representing both segmented and rbm addresses was specifically to avoid needing to do this kind of branch.  Update add_relative to behave correctly for rbm addresses as well.